lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -524,19 +544,42 @@ export default class BitcoinProcessor {
   }
 
   /**
-   * Return proof-of-fee value for a particular block.
+   * Modifies the given array and update the normalized fees, then write to block metadata store.
+   * @param blocks the ordered block metadata to set the normalized fee for.
    */
-  public async getNormalizedFee (block: number): Promise<TransactionFeeModel> {
+  private async writeBlocksToMetadataStoreWithFee (blocks: BlockMetadataWithoutNormalizedFee[]) {
+    for (const block of blocks) {
+      const normalizedFee = (await this.getNormalizedFee(block.height)).normalizedTransactionFee;
+      this.blockMetadataStore.add([{
```

Comment:
Maybe it's worthwhile to bulk insert just once at the end. (I say maybe 'coz I worry what would happen to CosmosDB rate limiting......)